src/set.cr

```diff
@@ -152,6 +152,30 @@ struct Set(T)
     set
   end
 
+  # Difference: returns a new set containing elements in this set that are not
+  # present in the other.
+  #
+  #     Set.new([1,2,3,4,5]) - Set.new([2,4])               #=> Set{1, 3, 5}
+  #     Set.new(['a','b','b','z']) - Set.new(['a','b','c']) #=> Set{'z'}
+  def -(other : Set)
+    set = Set(T).new
+    each do |value|
+      set.add value unless other.includes?(value)
+    end
+    set
+  end
```

Comment:
There's also `Set#subtract` to remove elements from a set without returning a new copy. I'd also like `Enumerable` as a restriction there.